modules/post/windows/capture/keylog_recorder.rb

```diff
@@ -87,116 +113,248 @@ def lock_screen
     end
   end
 
-  # Method to Migrate in to Explorer process to be able to interact with desktop
-  def process_migrate(captype,lock)
-    print_status("Migration type #{captype}")
-    #begin
-    if captype == "explorer"
-      process2mig = "explorer.exe"
-    elsif captype == "winlogon"
-      if is_uac_enabled? and not is_admin?
-        print_error("UAC is enabled on this host! Winlogon migration will be blocked.")
-
+  # This function returns the process name that the session is running in.
+  #
+  # Note: "session.sys.process[proc_name]" will not work when "include Msf::Post::Windows::Priv" is in the module.
+  #
+  # @return [String Class] the session process's name
+  # @return [NilClass] Session match was not found
+  def get_process_name
+    processes = client.sys.process.get_processes
+    processes.each do |proc|
+      if proc['pid'] == session.sys.process.getpid
```

Comment:
Call session.sys.process.getpid outside of the processes.each loop; you only need to call it once.